master/master.go

```diff
@@ -126,3 +141,38 @@ func instanceIds(instances []*ec2.Instance) []*string {
 	}
 	return instanceIds
 }
+
+func (m *Master) firstTaskIsValid() bool {
+	job := m.jobs[m.newTasks[0].JobID]
+	if job.reachedMaxTasks() {
+		m.requeueFirstTask()
+		return false
+	}
+	return true
+}
+
+func (m *Master) requeueFirstTask() {
+	t := m.newTasks[0]
+	m.newTasks = m.newTasks[1:]
+	m.newTasks = append(m.newTasks, t)
+}
+
+func (m *Master) slotsAvailable() bool {
+	for _, i := range m.instances {
+		if len(i.tasks) < i.maxTasks {
+			return true
+		}
+	}
+	return false
+}
+
+func (m *Master) scheduleTask() string {
```

Comment:
either the function name need to be changed (since it's simply returning an IP) or it's signature need to be changed.
perhaps change to `scheduleTask(t *Task)` and change the return statement to `SendTask(t, slaveIP)`